tests/unit/test_patch.py

```diff
@@ -0,0 +1,204 @@
+import json
+
+from dash import Patch
+from dash._utils import to_json
+
+
+def patch_to_dict(p):
+    return json.loads(to_json(p))
+
+
+def test_pat001_patch_assign_item():
+    p = Patch()
+    p["item"] = "item"
+
+    data = patch_to_dict(p)
+
+    assert data["operations"][0] == {
+        "operation": "Assign",
+        "location": ["item"],
+        "params": {"value": "item"},
+    }
+
+
+def test_pat002_patch_assign_attr():
+    p = Patch()
+    p.item = "item"
+
+    data = patch_to_dict(p)
+
+    assert data["operations"][0] == {
+        "operation": "Assign",
+        "location": ["item"],
+        "params": {"value": "item"},
+    }
+
+
+def test_pat003_patch_multi_operations():
+    p = Patch()
+    p.one = 1
+    p.two = 2
+
+    data = patch_to_dict(p)
+
+    assert len(data["operations"]) == 2
+    assert data["operations"][0]["location"] == ["one"]
+    assert data["operations"][1]["location"] == ["two"]
+
+
+def test_pat004_patch_nested_assign():
+    p = Patch()
+
+    p["nest_item"]["nested"]["deep"] = "deep"
+    p.nest_attr.nested.deep = "deep"
+
+    data = patch_to_dict(p)
+
+    assert data["operations"][0]["location"] == ["nest_item", "nested", "deep"]
+    assert data["operations"][1]["location"] == ["nest_attr", "nested", "deep"]
+
+
+def test_pat005_patch_delete_item():
+    p = Patch()
+
+    del p["delete_me"]
+
+    data = patch_to_dict(p)
+
+    assert data["operations"][0]["operation"] == "Delete"
+    assert data["operations"][0]["location"] == ["delete_me"]
+
+
+def test_pat006_patch_delete_attr():
+    p = Patch()
+
+    del p.delete_me
+
+    data = patch_to_dict(p)
+
+    assert data["operations"][0]["operation"] == "Delete"
+    assert data["operations"][0]["location"] == ["delete_me"]
+
+
+def test_pat007_patch_append():
+    p = Patch()
+    p.append("item")
+    data = patch_to_dict(p)
+
+    assert data["operations"][0] == {
+        "operation": "Append",
+        "location": [],
+        "params": {"value": "item"},
+    }
+
+
+def test_pat008_patch_prepend():
+    p = Patch()
+    p.prepend("item")
+    data = patch_to_dict(p)
+
+    assert data["operations"][0] == {
+        "operation": "Prepend",
+        "location": [],
+        "params": {"value": "item"},
+    }
+
+
+def test_pat009_patch_extend():
+    p = Patch()
+    p.extend(["extend"])
+    data = patch_to_dict(p)
+
+    assert data["operations"][0] == {
+        "operation": "Extend",
+        "location": [],
+        "params": {"value": ["extend"]},
+    }
+
+
+def test_pat010_patch_merge():
+    p = Patch()
+    p.merge({"merge": "merged"})
+    data = patch_to_dict(p)
+
+    assert data["operations"][0] == {
+        "operation": "Merge",
+        "location": [],
+        "params": {"value": {"merge": "merged"}},
+    }
+
+
+def test_pat011_patch_add():
+    p = Patch()
+    p.added = p.added + 1
```

Comment:
This is `__add__`, right? Do we actually want to allow this? Because it implies you could do something like `p.added = p.initial + 1` but that doesn't work, does it?